ui/lesson/OpCodeParser/OPFunctions.ts

```diff
@@ -0,0 +1,186 @@
+import { getKey, getSig, InterpreterRef } from '.'
+
+// Opcode types
+
+export const opCodeTypes = (opcode: string) => {
+  if (opcode.match(/OP_[0-9]*/)) {
+    return { opcode, operation: 'constants' }
+  }
+  switch (opcode) {
+    case 'OP_DUP':
+      return { opcode, operation: 'stack' }
+    case 'OP_ADD':
+      return { opcode, operation: 'arithmetic' }
+    case 'OP_EQUAL':
+      return { opcode, operation: 'bitwise' }
+    case 'OP_DUP':
+      return { opcode, operation: 'stack' }
+    case 'OP_DUP':
+      return { opcode, operation: 'stack' }
+    case 'OP_DUP':
+      return { opcode, operation: 'stack' }
+    default:
+      break
+  }
+}
```

Comment:
what does this do?

Reply:
So it should be able to get what type of operation each opcode code does

Reply:
why the repetition of `OP_DUP`?

Reply:
It's updated now, it was a test of how it would look for other Opcodes.